### DISCUSSION TITLE
Interested in joining forces? (emacs org-mode liberation)

### DISCUSSION THREAD

**Discussion by ngortheone** [guile-rs/guile-rs#19] [Development of guile-rs]
Hi @shymega @lidavidm 

I see that this repo is a bit abandoned, but I hope to revive your interest in it again.
I have an interesting idea/proposition for you.

I am author of https://github.com/org-rs/org-rs 
(also check out my [reddit post](https://old.reddit.com/r/rust/comments/bcz7d4/orgrs_org_parser_rewrite_in_rust/?st=jv8usn29&sh=94187a98))

The long standing goal of my project is to make org-mode available to non-emacs users.

While I am working on a parser rewrite in pure rust, it is only a small part of org-mode itself. There are thousands of lines of elisp code that will also need to be ported outside of emacs. Rewriting it all may take ages, unless a smarter approach is taken.

Since version 2.2 Guile claims to have 100% [compatibility with elisp](https://www.gnu.org/software/guile/manual/html_node/Using-Other-Languages.html) and this opens a unique opportunity of running unmodified elisp code outside of emacs.

I badly need a project like yours to make this possible. 
Will you be interested in continuing work on giule-rs and/or joining forces with org-rs organization?
We already have a couple of members and I am recruiting like-minded people.

Here are some examples of goals for guile-rs
- calling guile from rust and vice versa: emacs primitive funcitons,like cursor movements, text search etc will be implemented in rust, but have to be called from guile
- compiling rust statically linked with guile library - to be able to distribute the packaged code.

You are also welcome to take participation in org-rs directly, if you are interested. Rust is on the rise, and tools that follow unix philosophy shall rule the world!

**Comment by shymega** [comment]
Hello.

First off, thanks for reaching out! I'm so sorry for my late reply. Better late than never. I've been on holidays, and having laptop issues, which I'm still waiting for a replacement for, but I've got a spare running now....

Your project sounds intriguing, and I think this project could very well assist. I must explain however, that a) @lidavidm passed maintainership of `guile-rs` to me, although I haven't been able to work on it for a bit - aforementioned reasons, etc, and b) I should mention that [tim-editor/guile-rs](https://github.com/tim-editor/guile-rs) and this project are in the process of merging repos, to have a single, unified Rust binding to GNU Guile.

I see that @Jayvre has replied to your issue on the tim-editor/guile-rs repo as well.

Once we've figured out the approach for a) merging projects, b) interfacing with Guile via FFI, I would imagine that both Guile bindings will resume work.

**Comment by shymega** [comment]
Just for reference, this issue has been updated here -  tim-editor/guile-rs#9

**Comment by appetrosyan** [comment]
@ngortheone still interested?

**Comment by ngortheone** [comment]
@appetrosyan intellectually - yes, I still think it would be a cool project to pull off.
Realistically - I don't have the time these days :disappointed:
